jib-core/src/main/java/com/google/cloud/tools/jib/image/LayerEntry.java

```diff
@@ -52,10 +58,16 @@
    * @param sourceFile the source file to add to the layer
    * @param extractionPath the path in the container file system corresponding to the {@code
    *     sourceFile}
+   * @param permissions the file permissions on the container. Use {@code null} to use defaults (644
+   *     for files, 755 for directories)
    */
-  public LayerEntry(Path sourceFile, AbsoluteUnixPath extractionPath) {
+  public LayerEntry(
+      Path sourceFile,
+      AbsoluteUnixPath extractionPath,
+      @Nullable Set<PosixFilePermission> permissions) {
```

Comment:
Would rather just convert it to an `ImmutableSet` within the constructor instead of wrapping the parameter in an `ImmutableSet.copyOf` every time we call the constructor throughout the codebase.